src/libs/actions/PaymentMethods.js

```diff
@@ -107,7 +107,54 @@ function openPaymentsPage() {
         ],
     };
 
-    return API.read('OpenPaymentsPage', {}, onyxData);
+    return API.read('OpenPaymentsPage', {
+        // We're passing this to have the data returned in the right format.
+        // This can be removed when the massageData parameter
+        // is removed from here https://github.com/Expensify/Web-Expensify/blob/main/lib/BankAccountAPI.php#L1064.
+        massageData: true,
+    }, onyxData);
+}
+
+/**
+ *
+ * @param {Number} bankAccountID
+ * @param {Number} fundID
+ * @param {Object} previousPaymentMethod
+ * @param {Object} currentPaymentMethod
+ * @param {Boolean} isOptimisticData
+ * @return {Array}
+ *
+ */
+function getMakeDefaultPaymentOnyxData(bankAccountID, fundID, previousPaymentMethod, currentPaymentMethod, isOptimisticData = true) {
+    return [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.USER_WALLET,
+            value: {
+                walletLinkedAccountID: bankAccountID || fundID,
+                walletLinkedAccountType: bankAccountID ? CONST.PAYMENT_METHODS.BANK_ACCOUNT : CONST.PAYMENT_METHODS.DEBIT_CARD,
+                errors: null,
```

Comment:
This line is clearing the error in `ONYXKEYS.USER_WALLET`, and we are using it as success and failure data. This ends up clearing the error a few milliseconds after getting it from the server.


https://user-images.githubusercontent.com/87341702/200662747-dad05016-8ccf-4684-8f2a-15d47114fc92.mov

See the error at the bottom popping and disappearing almost immediatelly.

Reply:
Oof, thanks for catching that, @aldo-expensify. PR fix here: https://github.com/Expensify/App/pull/12591

Reply:
Sorry @MariaHCD, I didn't link it in this comment, but I created a GH issue and a PR is already under review for that: https://github.com/Expensify/App/pull/12579